File: exercises/exercism.io/python/bob/bob.py
class Bob(object):

  def hey(self, phrase):
    utterance = Utterance(phrase)

    if utterance.is_silence():
        return 'Fine. Be that way!'
    elif utterance.is_a_shout():
        return 'Woah, chill out!'
    elif utterance.is_a_question():
        return 'Sure.'
    else:
        return 'Whatever.'


class Utterance(object):

  def __init__(self, phrase):
    self.phrase = phrase

  def is_silence(self):
    return not self.phrase.strip()

  def is_a_question(self):
    return self.phrase.endswith('?')

  def is_a_shout(self):
    return self.phrase.isupper()

File: exercises/exercism.io/go/bob/bob.go
package bob

import (
	"regexp"
	"strings"
)

func Hey(remark string) string {
	remark = strings.TrimSpace(remark)

	if isAYell(remark) && isAQuestion(remark) {
		return "Calm down, I know what I'm doing!"
	} else if isAYell(remark) {
		return "Whoa, chill out!"
	} else if isAQuestion(remark) {
		return "Sure."
	} else if isInSilence(remark) {
		return "Fine. Be that way!"
	} else {
		return "Whatever."
	}
}

func isAYell(remark string) bool {
	hasLetters, _ := regexp.MatchString("[A-Z]", remark)
	return hasLetters && remark == strings.ToUpper(remark)
}

func isAQuestion(remark string) bool {
	return strings.HasSuffix(remark, "?")
}

func isInSilence(remark string) bool {
	return remark == ""
}
